MTD simulation: ETL update

casarsa opened the pull request:
#### PR description:

This is an update of the ETL simulation. The contents of this PR were presented at the MTD DPG Meeting on 15/05/2020 ([slides](https://indico.cern.ch/event/919571/contributions/3865744/attachments/2039827/3415899/casarsa_simETL.pdf)).
In summary, the changes are:
- a small fix to the saturation of ETL ADC and TDC counts;
- the energy of DIGI hits is now expressed in fC instead of MIP units;
- a more realistic parameterization of the ETL resolution is introduced as a function of the LGAD sensor gain, which depends on the fluence at the sensor radial position. 

----------------------------------------------------------------------------------------------------------

#### PR validation:

The new code has been run in CMSSW_11_1_X_2020-05-17-2300 for the WFs 23434.0 and 23434.99.


**Memory and time reports for WF 23434.0**

step 2:

MemoryReport> Peak virtual size 5611.59 Mbytes                                                                                    
 Key events increasing vsize:                                                                                                     
[0] run: 0 lumi: 0 event: 0  vsize = 0 deltaVsize = 0 rss = 0 delta = 0                                                           
[1] run: 1 lumi: 1 event: 1  vsize = 5099.55 deltaVsize = 0 rss = 3257.18 delta = 0                                               
[2] run: 1 lumi: 1 event: 2  vsize = 5611.55 deltaVsize = 512.004 rss = 3227.38 delta = -29.8047                                  
[3] run: 1 lumi: 1 event: 3  vsize = 5611.58 deltaVsize = 0.0234375 rss = 3361.9 delta = 134.527                                  
[6] run: 1 lumi: 1 event: 6  vsize = 5611.59 deltaVsize = 0.0078125 rss = 3244.89 delta = -117.012                                
[8] run: 1 lumi: 1 event: 8  vsize = 5611.59 deltaVsize = 0 rss = 3337.38 delta = 92.4922                                         
[7] run: 1 lumi: 1 event: 7  vsize = 5611.59 deltaVsize = 0 rss = 3220.32 delta = -24.5664                                        
[6] run: 1 lumi: 1 event: 6  vsize = 5611.59 deltaVsize = 0.0078125 rss = 3244.89 delta = -117.012                                
TimeReport> Time report complete in 1298.63 seconds                                                                               
 Time Summary:                                                                                                                    
 - Min event:   110.279                                                                                                           
 - Max event:   129.325                                                                                                           
 - Avg event:   118.313                                                                                                           
 - Total loop:  1265.45                                                                                                           
 - Total init:  32.6732                                                                                                           
 - Total job:   1298.63                                                                                                           
 - EventSetup Lock:   0.000230551                                                                                                 
 - EventSetup Get:   140.65                                                                                                       
 Event Throughput: 0.0079023 ev/s                                                                                                 
 CPU Summary:                                                                                                                     
 - Total loop:  1206.82                                                                                                           
 - Total init:  15.8521                                                                                                           
 - Total extra: 0                                                                                                                 
 - Total job:   1223.1                                                                                                            
 Processing Summary:                                                                                                              
 - Number of Events:  10                                                                                                          
 - Number of Global Begin Lumi Calls:  1                                                                                          
 - Number of Global Begin Run Calls: 1 

step 3:

MemoryReport> Peak virtual size 11848.2 Mbytes                                                                                    
 Key events increasing vsize:                                                                                                     
[1] run: 1 lumi: 1 event: 1  vsize = 10789 deltaVsize = 0 rss = 7388.9 delta = 0                                                  
[2] run: 1 lumi: 1 event: 2  vsize = 10789 deltaVsize = 0.0234375 rss = 7227.34 delta = -161.562                                  
[3] run: 1 lumi: 1 event: 3  vsize = 11813.2 deltaVsize = 1024.16 rss = 8005.51 delta = 778.172                                   
[6] run: 1 lumi: 1 event: 6  vsize = 11845.2 deltaVsize = 32 rss = 7514.96 delta = -490.551                                       
[8] run: 1 lumi: 1 event: 8  vsize = 11848.2 deltaVsize = 3 rss = 7921.5 delta = 406.539                                          
[9] run: 1 lumi: 1 event: 9  vsize = 11847.2 deltaVsize = -1 rss = 7731.43 delta = -190.066                                       
[10] run: 1 lumi: 1 event: 10  vsize = 11847.2 deltaVsize = -0.984375 rss = 7768.05 delta = -153.453                              
[8] run: 1 lumi: 1 event: 8  vsize = 11848.2 deltaVsize = 3 rss = 7921.5 delta = 406.539                                          
TimeReport> Time report complete in 3943.92 seconds                                                                               
 Time Summary:                                                                                                                    
 - Min event:   269.565                                                                                                           
 - Max event:   489.927                                                                                                           
 - Avg event:   360.594                                                                                                           
 - Total loop:  3736.06                                                                                                           
 - Total init:  195.717                                                                                                           
 - Total job:   3943.92                                                                                                           
 - EventSetup Lock:   0.00048089                                                                                                  
 - EventSetup Get:   211.613                                                                                                      
 Event Throughput: 0.00267662 ev/s                                                                                                
 CPU Summary:                                                                                                                     
 - Total loop:  3525.64                                                                                                           
 - Total init:  153.563                                                                                                           
 - Total extra: 0                                                                                                                 
 - Total job:   3691.01                                                                                                           
 Processing Summary:                                                                                                              
 - Number of Events:  10                                                                                                          
 - Number of Global Begin Lumi Calls:  1                                                                                          
 - Number of Global Begin Run Calls: 1 


**Memory and time reports for WF 23434.99**

step 2:

MemoryReport> Peak virtual size 10251.1 Mbytes                                                                                    
 Key events increasing vsize:                                                                                                     
[0] run: 0 lumi: 0 event: 0  vsize = 0 deltaVsize = 0 rss = 0 delta = 0                                                           
[1] run: 1 lumi: 1 event: 1  vsize = 8459.01 deltaVsize = 0 rss = 3910.97 delta = 0                                               
[2] run: 1 lumi: 1 event: 2  vsize = 10251 deltaVsize = 1792.02 rss = 4521.15 delta = 610.18                                      
[3] run: 1 lumi: 1 event: 3  vsize = 10251.1 deltaVsize = 0.0273438 rss = 5960.33 delta = 1439.18                                 
[0] run: 0 lumi: 0 event: 0  vsize = 0 deltaVsize = 0 rss = 0 delta = 0                                                           
[5] run: 1 lumi: 1 event: 5  vsize = 10251.1 deltaVsize = 0 rss = 4283.99 delta = -1676.34                                        
[4] run: 1 lumi: 1 event: 4  vsize = 10251.1 deltaVsize = 0 rss = 5861.37 delta = -98.957                                         
[3] run: 1 lumi: 1 event: 3  vsize = 10251.1 deltaVsize = 0.0273438 rss = 5960.33 delta = 1439.18                                 
TimeReport> Time report complete in 1155.37 seconds                                                                               
 Time Summary:                                                                                                                    
 - Min event:   88.3925                                                                                                           
 - Max event:   113.659                                                                                                           
 - Avg event:   95.0481                                                                                                           
 - Total loop:  1121.69                                                                                                           
 - Total init:  33.4398                                                                                                           
 - Total job:   1155.37                                                                                                           
 - EventSetup Lock:   0.00013113                                                                                                  
 - EventSetup Get:   149.902                                                                                                      
 Event Throughput: 0.00891514 ev/s                                                                                                
 CPU Summary:                                                                                                                     
 - Total loop:  1070.38                                                                                                           
 - Total init:  11.5136                                                                                                           
 - Total extra: 0                                                                                                                 
 - Total job:   1082.07                                                                                                           
 Processing Summary:                                                                                                              
 - Number of Events:  10                                                                                                          
 - Number of Global Begin Lumi Calls:  1                                                                                          
 - Number of Global Begin Run Calls: 1    

step 3:

MemoryReport> Peak virtual size 7945.79 Mbytes                                                                                    
 Key events increasing vsize:                                                                                                     
[0] run: 0 lumi: 0 event: 0  vsize = 0 deltaVsize = 0 rss = 0 delta = 0                                                           
[1] run: 1 lumi: 1 event: 1  vsize = 7483.72 deltaVsize = 0 rss = 4517.23 delta = 0                                               
[2] run: 1 lumi: 1 event: 2  vsize = 7945.72 deltaVsize = 462 rss = 4449.54 delta = -67.6914                                      
[4] run: 1 lumi: 1 event: 4  vsize = 7945.76 deltaVsize = 0.0351562 rss = 4523.5 delta = 73.957                                   
[8] run: 1 lumi: 1 event: 8  vsize = 7945.77 deltaVsize = 0.015625 rss = 4523.7 delta = 0.203125                                  
[8] run: 1 lumi: 1 event: 8  vsize = 7945.77 deltaVsize = 0.015625 rss = 4523.7 delta = 0.203125                                  
[9] run: 1 lumi: 1 event: 9  vsize = 7945.79 deltaVsize = 0.0117188 rss = 4427.04 delta = -96.6562                                
[10] run: 1 lumi: 1 event: 10  vsize = 7945.79 deltaVsize = 0.0078125 rss = 4572.27 delta = 145.223                               
TimeReport> Time report complete in 724.497 seconds                                                                               
 Time Summary:                                                                                                                    
 - Min event:   57.2656                                                                                                           
 - Max event:   71.7829                                                                                                           
 - Avg event:   62.375                                                                                                            
 - Total loop:  703.033                                                                                                           
 - Total init:  20.9152                                                                                                           
 - Total job:   724.497                                                                                                           
 - EventSetup Lock:   0.000170231                                                                                                 
 - EventSetup Get:   146.07                                                                                                       
 Event Throughput: 0.0142241 ev/s                                                                                                 
 CPU Summary:                                                                                                                     
 - Total loop:  661.372                                                                                                           
 - Total init:  15.7546                                                                                                           
 - Total extra: 0                                                                                                                 
 - Total job:   677.598                                                                                                           
 Processing Summary:                                                                                                              
 - Number of Events:  10                                                                                                          
 - Number of Global Begin Lumi Calls:  1                                                                                          
 - Number of Global Begin Run Calls: 1  

step 4:

MemoryReport> Peak virtual size 6853.92 Mbytes                                                                                    
 Key events increasing vsize:                                                                                                     
[0] run: 0 lumi: 0 event: 0  vsize = 0 deltaVsize = 0 rss = 0 delta = 0                                                           
[1] run: 1 lumi: 1 event: 1  vsize = 6181.48 deltaVsize = 0 rss = 4517.5 delta = 0                                                
[2] run: 1 lumi: 1 event: 2  vsize = 6821.49 deltaVsize = 640.016 rss = 4441.75 delta = -75.75                                    
[6] run: 1 lumi: 1 event: 6  vsize = 6853.63 deltaVsize = 32 rss = 4492.17 delta = 50.4141                                        
[10] run: 1 lumi: 1 event: 10  vsize = 6853.92 deltaVsize = 0.238281 rss = 4574.88 delta = 82.707                                 
[6] run: 1 lumi: 1 event: 6  vsize = 6853.63 deltaVsize = 32 rss = 4492.17 delta = -23.4961                                       
[9] run: 1 lumi: 1 event: 9  vsize = 6853.68 deltaVsize = 0.0507812 rss = 4500.19 delta = 8.01953                                 
[10] run: 1 lumi: 1 event: 10  vsize = 6853.92 deltaVsize = 0.238281 rss = 4574.88 delta = 74.6875                                
TimeReport> Time report complete in 2464.96 seconds                                                                               
 Time Summary:                                                                                                                    
 - Min event:   192.316                                                                                                           
 - Max event:   253.524                                                                                                           
 - Avg event:   217.986                                                                                                           
 - Total loop:  2288.57                                                                                                           
 - Total init:  164.638                                                                                                           
 - Total job:   2464.96                                                                                                           
 - EventSetup Lock:   0.000385284                                                                                                 
 - EventSetup Get:   207.967                                                                                                      
 Event Throughput: 0.00436954 ev/s                                                                                                
 CPU Summary:                                                                                                                     
 - Total loop:  2212.66                                                                                                           
 - Total init:  148.033                                                                                                           
 - Total extra: 0                                                                                                                 
 - Total job:   2372.13                                                                                                           
 Processing Summary:                                                                                                              
 - Number of Events:  10                                                                                                          
 - Number of Global Begin Lumi Calls:  1                                                                                          
 - Number of Global Begin Run Calls: 1

casarsa committed:
Fixed ADC and TDC saturation for ETL

casarsa committed:
Energy of ETL DIGIs converted from MIP units to femtocoulomb

casarsa committed:
Implemented the dependence of the ETL time resolution on the LGAD gain, which is a function of the fluence at the sensor's position

casarsa committed:
Updated the energy calibration constant for the ETL RECO hits.

casarsa committed:
Extended ETL ADC range

casarsa committed:
ETL time resolution changed into a FormulaEvaluator

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-29898/15479

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  RecoLocalFastTime/FTLRecProducers/python/mtdRecHits_cfi.py  modified in PR(s): #29803
  - File  RecoLocalFastTime/FTLRecProducers/python/mtdUncalibratedRecHits_cfi.py  modified in PR(s): #29803

cmsbuild commented:
A new Pull Request was created by  @casarsa (Massimo Casarsa) for master.

It involves the following packages:

RecoLocalFastTime/FTLRecProducers
SimFastTiming/FastTimingCommon


@perrotta, @civanch, @kpedro88, @cmsbuild, @mdhildreth, @slava77 can you please review it and eventually sign? Thanks.
@fabiocos this is something you requested to watch as well.
@silviodonato, @dpiparo you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

fabiocos commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ac5c2a02aaf3e8a3eb8bf4fcd4a74f8f13bb47dc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e718a/6396/summary.html
CMSSW: CMSSW_11_1_X_2020-05-18-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e718a/6396/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2648 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2694466
 - DQMHistoTests: Total failures: 28451
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2665966
 - DQMHistoTests: Total skipped: 49
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 150 log files, 16 edm output root files, 35 DQM output files

slava77 commented:
Looking at the comparisons, I'm not sure that everything is as expected; e.g. in wf 23234 (D49 ttbar, no PU)

in mtdRechits it looks like there is a truncation on energy at 1.0
![all_OldVSNew_TTbar14TeV2026D49wf23234p0c_log10FTLRecHitsSorted_mtdRecHits_FTLEndcap_RECO_obj_obj_energy](https://user-images.githubusercontent.com/4676718/82347390-62d87900-99ac-11ea-8338-33a630324384.png)

<img width="740" alt="wf23234_ETL_hits_E" src="https://user-images.githubusercontent.com/4676718/82347975-235e5c80-99ad-11ea-9de0-a7691e3b4e6a.png">

This seems to match an overflow entry in the digis
<img width="738" alt="wf23234_ETL_digi_-z_Qadc" src="https://user-images.githubusercontent.com/4676718/82348038-340ed280-99ad-11ea-9464-40fffcc1811a.png">

Also, the clusters now all have more than 1 hit (but the stats are rather small)
<img width="741" alt="wf23234_ETL_hitsCluster" src="https://user-images.githubusercontent.com/4676718/82348130-4f79dd80-99ad-11ea-9c72-c260c912b603.png">

@casarsa 
I see that the saturation values changed, which I suppose corresponds to the new max values and the "truncation" in the hit energy.
However, I wasn't expecting so much saturation in this no-pileup test.
I'm not sure which change connects to the change in the cluster size.
Is this expected? Please point me to some plots on this.
Thank you.

casarsa commented:
@slava77 
The change in the energy spectrum of the ETL RecHits is expected. The saturation peak at 1 MeV comes from the saturation of the ADC counts of the DIGIs, that you've noticed.

In the DIGIs' charge plot there are two expected changes. I have fixed the ADC saturation and I've changed the units: before the DIGI charge was expressed in MIP units, now it's convented to fC.

As for the cluster size, I think that the discrepancy is due to the very low statistics. I've rechecked the plot with 100 TTbar events (WF 23234):

![ETL_ClusterSize_23234](https://user-images.githubusercontent.com/5435977/82371347-5428ab00-9a1a-11ea-85c1-0d26ff92e305.png)

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/29898	ac5c2a0
- code changes are in line with the PR description; changes in RECO parts of the code are rather straightforward, to change some calibration/saturation constants in sync with the DIGI side
- jenkins tests pass and comparisons with the baseline show differences in the phase-2 workflows; due to changes at the digi level random numbers the differences spread to other detectors and only the most significantly changing plots are easy to understand on the low number of events in the tests (see discussion above)

kpedro88 commented:
please test workflows 23234.1001,23434.1001

cmsbuild commented:
The tests are being triggered in jenkins.
Test Parameters:
  - **MATRIX_EXTRAS** = 23434.1001,23234.1001

cmsbuild commented:
+1
Tested at: ac5c2a02aaf3e8a3eb8bf4fcd4a74f8f13bb47dc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e718a/6446/summary.html
CMSSW: CMSSW_11_1_X_2020-05-19-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e718a/6446/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-2e718a/23234.1001_TTbar_14TeV+RecoFullGlobal_TestOldDigi_2026D49+HARVESTFullGlobal_2026D49
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-2e718a/23434.1001_TTbar_14TeV+RecoFullGlobalPU_TestOldDigi_2026D49PU+HARVESTFullGlobalPU_2026D49PU

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2652 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2694466
 - DQMHistoTests: Total failures: 28446
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2665971
 - DQMHistoTests: Total skipped: 49
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 150 log files, 16 edm output root files, 35 DQM output files

civanch commented:
+1

kpedro88 commented on RecoLocalFastTime/FTLRecProducers/python/mtdRecHits_cfi.py:
```diff
@@ -9,8 +9,8 @@
-    thresholdToKeep = cms.double(0.0425),    # MeV
-    calibrationConstant = cms.double(0.085), # MeV/MIP
+    thresholdToKeep = cms.double(0.0425),       # MeV
+    calibrationConstant = cms.double(0.023156), # MeV/fC
```
I'm concerned this change could interfere with processing 11_0_X DIGI files for the TDR re-reco. Since those digis were simulated with the old calibration, using the new calibration in RECO could produce incorrect results. It might be better to hold this PR until 11_2_X.

slava77 replied:
good point (and I forgot about the 11_0_X compatibility point).
I agree with the hold.
However, IIRC in the last ORP the proposal was to extend this compatibility range to 11_2_X as well.
Should we put a proc modifier here and in digi?

fabiocos replied:
It should interfere IMO, if only a reRECO is done. This could would allow a study of luminosity effects, but only if DIGI are reproduced again, which is not in the plan. A proc modifier might allow us to use this on newer versions of the detector for studies without waiting winter. I am not sure anyway how much more complex the could would become, @casarsa what do you think?

casarsa replied:
I would roll back the conversion of energy to MIP units. The focus of this PR is on the time resolution. The conversion to fC is just an intermediate step, not essential at this stage. I was planning in advance and setting up things for a future and longer-term development when a pulse shape of the electronic signal is available.

kpedro88 commented:
hold

cmsbuild commented:
Pull request has been put on hold by  @kpedro88
They need to issue an `unhold` command to remove the `hold` state or L1 can `unhold` it for all

kpedro88 commented:
please test
w/ #29942 now merged, we should see differences from the IB in the 11_0_X test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ac5c2a02aaf3e8a3eb8bf4fcd4a74f8f13bb47dc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e718a/6518/summary.html
CMSSW: CMSSW_11_1_X_2020-05-22-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e718a/6518/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2649 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2780792
 - DQMHistoTests: Total failures: 28577
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2752165
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

fabiocos commented:
as expected only ETL and its impact on the vertex reconstruction are affected...

slava77 commented:
@cmsbuild please test

to pick up reco comparisons for the workflow .1001

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ac5c2a02aaf3e8a3eb8bf4fcd4a74f8f13bb47dc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e718a/6568/summary.html
CMSSW: CMSSW_11_2_X_2020-05-24-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e718a/6568/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2645 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2780792
 - DQMHistoTests: Total failures: 28576
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2752166
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

silviodonato commented:
@casarsa are you going to revert the change pointed by @kpedro88 https://github.com/cms-sw/cmssw/pull/29898#discussion_r428793142 ?
Since it is matter of re-HLT of HLT TDR samples, let me add @fwyzard @trtomei in cc.
I think it can be merged, but we need a customization function for the HLT to run on 110X samples.
In any case, it is even better to keep this change in a separate PR

casarsa commented:
@silviodonato yes, I'm going to revert the energy conversion to fC. I'm running some checks now and then I will push the updatd branch

casarsa committed:
Reverted the conversion of ETL hit energy to fC for backward compatibility.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-29898/15673

- This PR adds an extra 20KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  SimFastTiming/FastTimingCommon/src/ETLElectronicsSim.cc  modified in PR(s): #29935


Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-29898/15673/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-29898/15673/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

casarsa committed:
Applied code-format

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-29898/15674

- This PR adds an extra 20KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  SimFastTiming/FastTimingCommon/src/ETLElectronicsSim.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #29898 was updated. @Martin-Grunewald, @civanch, @Dr15Jones, @makortel, @cvuosalo, @ianna, @mdhildreth, @cmsbuild, @kpedro88, @fwyzard can you please check and sign again.

fabiocos commented:
@casarsa you are adding some external PR in your branch, please rebase it properly on the latest IB.

casarsa commented:
I've noticed. But the branch has already been rebased on CMSSW_11_1_X_2020-05-26-2300.

perrotta commented:
> I've noticed. But the branch has already been rebased on CMSSW_11_1_X_2020-05-26-2300.

Master has now moved to 11_2_X: please rebace on CMSSW_11_2_X_2020-05-26-2300

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-29898/15682

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  SimFastTiming/FastTimingCommon/src/ETLElectronicsSim.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #29898 was updated. @cmsbuild, @civanch, @kpedro88, @mdhildreth can you please check and sign again.

fabiocos commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 469225aa6eba285ec360daa4ee911fe19bc1e5bb
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e718a/6622/summary.html
CMSSW: CMSSW_11_2_X_2020-05-27-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2e718a/6622/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 223 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2780792
 - DQMHistoTests: Total failures: 579
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2780163
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

kpedro88 commented:
unhold
changes in WF 23234.1001 no longer appear

kpedro88 commented:
(very minor) comparison changes in phase 2 workflows arise in ETL digis:
![ETL digis](http://kjplanet.com/pr/MTD__ETL_DigiHits_EtlHitTvsPhiZneg.png)

and propagate to ETL-related reco quantities:
![ETL cluster energy](http://kjplanet.com/pr/all_OldVSNew_TTbar14TeV2026D49wf23234p0c_FTLClusteredmNewDetSetVector_mtdClusters_FTLEndcap_RECO_obj_m_data_energy.png)

Given the nature of this PR, I assume this is expected, but please confirm @casarsa

casarsa commented:
@kpedro88 Yes, I confirm

kpedro88 commented:
+upgrade

silviodonato commented:
Do you have any comments? simulation:  @civanch @mdhildreth

civanch commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
+1
